File: 704-binary-search/704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        # how to do binary search correctly?
        # I i were to insert target into nums, what index would I insert it in?
        l = 0
        r = len(nums)
        while l < r:
            m = (l + r) // 2
            if nums[m] == target:
                return m
            if nums[m] < target:
                l = m + 1
            else:
                r = m
        return -1
    

File: 704-binary-search/704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length;
        while (l < r) {
            int m = (l + r) / 2;
            if (nums[m] == target) {
                return m;
            }
            if (target < nums[m]) {
                r = m;
            } else {
                l = m + 1;
            }
        }
        return -1;
    }
}